pkg/crc/cache/cache.go

```diff
@@ -69,12 +74,17 @@ func (c *Cache) CacheBinary() error {
 	if err != nil {
 		return err
 	}
-
-	// Extract the tarball and put it the cache directory.
-	extractedFiles, err := extract.UncompressWithFilter(assetTmpFile, tmpDir,
-		func(filename string) bool { return filepath.Base(filename) == c.binaryName })
-	if err != nil {
-		return errors.Wrapf(err, "Cannot uncompress '%s'", assetTmpFile)
+	var extractedFiles []string
+	// Check the file is tarball or not
+	if IsTarball(assetTmpFile) {
+		// Extract the tarball and put it the cache directory.
+		extractedFiles, err = extract.UncompressWithFilter(assetTmpFile, tmpDir,
```

Comment:
Let say I don't have oc and goodhost on my machine, will it unzip two times the archive ?

Reply:
@guillaumerose no it shouldn't extract it twice, this logic check if the file is a tarball type or not if yes the only it will run the extract otherwise just download.

Reply:
oh thanks! I thought it would extract the crcbundle but it's in fact the archive containing the bits in the upstream mirror.